lib/kafo/puppet_configurer.rb

```diff
@@ -23,15 +19,13 @@ def []=(key, value)
     end
 
     def write_config
-      @logger.debug("Writing Puppet config file at #{@temp_file.path}")
-      @temp_file.open
-      @temp_file.truncate(0)
-      @temp_file.puts '[main]'
-      @settings.keys.sort.each do |key|
-        @temp_file.puts "#{key} = #{@settings[key]}"
+      @logger.debug("Writing Puppet config file at #{config_path}")
+      File.open(config_path, 'w') do |f|
```

Comment:
I'm less of a fan on multiline blocks of using a single character variable as it hurts readability of the code when something like `file` provides more context.